pkg/controller/podautoscaler/monitor/metrics.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// metrics packages contains metrics which are exposed from the HPA controller.
+package monitor
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	// hpaControllerSubsystem - subsystem name used by HPA controller
+	hpaControllerSubsystem = "hpa_controller"
```

Comment:
There's currently [efforts in other areas](https://github.com/kubernetes/kubernetes/pull/115813/files#diff-30acbb8d2e7a2aff26694443ae9353e09061f9a5aeb0f2703960371dda2e4e5cR65) to standardise the controller name scheme, with the HPA controller being referred to as `horizontal-pod-autoscaler-controller` there. Better to start with consistency with the proposed standard going forward unless there's a strong reason not to?

Reply:
👍 
I'll rename here to `"horizontal_pod_autoscaler_controller"` in https://github.com/kubernetes/kubernetes/pull/116010. (replace `-` with `_` as no other metric named with `-`. ([ref](https://kubernetes.io/docs/reference/instrumentation/metrics/)))